services/QuillLMS/engines/evidence/app/services/evidence/research/gen_ai/dataset_editor/changes_detector.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module Evidence
+  module Research
+    module GenAi
+      module DatasetEditor
+        class ChangesDetector < ApplicationService
+          attr_reader :dataset, :changes, :klass, :version_type
+
+          def initialize(dataset:, changes:, klass:, version_type:)
+            @dataset = dataset
+            @changes = changes
+            @klass = klass
+            @version_type = version_type
+          end
+
+          def run = !!deleted_changes? || !!created_changes? || !!updated_changes?
```

Comment:
Maybe move the !! only to where it's needed (explicit booleans are returned in a lot of places) to make this more readable, i.e. `deleted_changes? || created_changes? || updated_changes?`